packages/did-comm/src/didcomm.ts

```diff
@@ -494,15 +496,60 @@ export class DIDComm implements IAgentPlugin {
     return { msgObj, mediaType }
   }
 
-  private findPreferredDIDCommService(services: any) {
+  private findPreferredDIDCommService(services: Service[]) {
     // FIXME: TODO: get preferred service endpoint according to configuration; now defaulting to first service
     return services[0]
   }
 
+  private async wrapDIDCommForwardMessage(
+    recipientDidUrl: string,
+    messageId: string,
+    packedMessageToForward: IPackedDIDCommMessage,
+    routingKey: string,
+    context: IAgentContext<IDIDManager & IKeyManager & IResolver>,
+  ): Promise<IPackedDIDCommMessage> {
+    const splitKey = routingKey.split('#')
+    const shouldUseSpecificKid = splitKey.length > 1
+    const mediatorDidUrl = splitKey[0]
+    // 1. Create forward message
+    const forwardMessage: IDIDCommMessage = {
+      id: uuidv4(),
+      type: 'https://didcomm.org/routing/2.0/forward',
+      to: mediatorDidUrl,
+      body: {
+        next: recipientDidUrl,
+      },
+      attachments: [
+        {
+          media_type: DIDCommMessageMediaType.ENCRYPTED,
+          data: {
+            json: JSON.parse(packedMessageToForward.message),
+          },
+        },
+      ],
+    }
+
+    context.agent.emit('DIDCommV2Message-forwarded', {
+      messageId,
+      next: recipientDidUrl,
+      routingKey: routingKey,
+    })
+
+    // 2. Pack message for routingKey with anoncrypt
+    if (shouldUseSpecificKid) {
+      return this.packDIDCommMessageJWE(
+        { message: forwardMessage, packing: 'anoncrypt', options: { recipientKids: [routingKey] } },
```

Comment:
does the forward protocol assume anoncrypt?

Reply:
I don't see anywhere in the spec where this is mentioned explicitly. The example does not have a `from` field, meaning anoncrypt is used. https://identity.foundation/didcomm-messaging/spec/#messages

This seems to mean either could be used?